src/libstore/profiles.cc

```diff
@@ -305,5 +308,13 @@ Path getDefaultProfile()
     }
 }
 
+void createLegacyProfileLink(Path oldProfile, Path newProfile)
+{
+    if (pathExists(oldProfile))
+        return;
+    createDirs(dirOf(oldProfile));
+    replaceSymlink(newProfile, oldProfile);
+}
```

Comment:
Make naming consistent with the rest of PR, and less confusing

```suggestion
void createLegacyProfilesLink(Path oldProfilesDir, Path newProfilesDir)
{
    if (pathExists(oldProfilesDir))
        return;
    createDirs(dirOf(oldProfilesDir));
    replaceSymlink(newProfilesDir, oldProfilesDir);
}
```